orchestration/dap_orchestration/solids.py

```diff
@@ -1,15 +1,40 @@
-from typing import Union
+from typing import Any, Union
 
-from dagster import Bool, Failure, String, solid, configured, InputDefinition, Any, Tuple, OutputDefinition
+from dagster import Bool, Failure, String, solid, configured, InputDefinition, Nothing
 from dagster.core.execution.context.compute import AbstractComputeExecutionContext
-from dagster_utils.contrib.google import gs_path_from_bucket_prefix, parse_gs_path, path_has_any_data
+from dagster_utils.contrib.google import parse_gs_path
+
 from dap_orchestration.types import DapSurveyType, FanInResultsWithTsvDir
 
 extract_project = "dog-aging-hles-extraction"
 transform_project = "dog-aging-hles-transformation"
 class_prefix = "org.broadinstitute.monster.dap"
 
 
+@solid(required_resource_keys={"slack_client"})
+def send_pipeline_start_notification(context: AbstractComputeExecutionContext) -> None:
+    message = [
+        "Dog Aging Pipeline starting",
+        f"Name = {context.pipeline_def.name}",
+        f"Run ID = {context.run_id}"
+    ]
+    context.resources.slack_client.send_message("\n".join(message))
+
+
+@solid(
+    required_resource_keys={"slack_client"},
+    input_defs=[InputDefinition("ignore1"), InputDefinition("ignore2")]
+)
+def send_pipeline_finish_notification(
```

Comment:
we wouldn't need a separate solid for failures since the make_slack_on_pipeline_failure_sensor will handle those cases, correct?